src/renderer/components/cluster-manager/cluster-view.tsx

```diff
@@ -26,15 +27,37 @@ export class ClusterView extends React.Component<Props> {
 
   async componentDidMount() {
     disposeOnUnmount(this, [
-      reaction(() => this.clusterId, clusterId => ClusterStore.getInstance().setActive(clusterId), {
+      reaction(() => this.clusterId, (clusterId) => {
+        this.showCluster(clusterId);
+      }, {
         fireImmediately: true,
-      }),
-      reaction(() => this.cluster.online, (online) => {
-        if (!online) navigate(catalogURL());
       })
     ]);
   }
 
+  componentWillUnmount() {
+    this.hideCluster();
+  }
+
+  showCluster(clusterId: string) {
+    initView(clusterId);
+    requestMain(clusterActivateHandler, this.clusterId, false);
+
+    const entity = catalogEntityRegistry.getById(this.clusterId);
+
+    if (entity) {
+      catalogEntityRegistry.activeEntity = entity;
+    }
```

Comment:
Why would we not revert to the catalog in the invalid ID state?

Reply:
Good question, we probably should. I'll this (and few other things) in a follow-up PR.